sageopt/tests/test_coniclifts/test_problem.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+import numpy as np
+import time
+from sageopt.coniclifts.standards import constants as CL_CONSTANTS
+from sageopt.coniclifts.problems.solvers.ecos import ECOS
+from sageopt.coniclifts.problems.solvers.mosek import Mosek
+from sageopt.coniclifts.compilers import compile_problem
+from sageopt.coniclifts.base import Expression, Variable
+from sageopt.coniclifts import Problem, Constraint
+import sageopt.coniclifts as cl
+
+class TestProblem(unittest.TestCase):
+    def test_solve(self):
+        # random problem data
+        G = np.random.randn(3, 6)
+        h = G @ np.random.rand(6)
+        c = np.random.rand(6)
+        # input to coniclift's Problem constructor
+        x = cl.Variable(shape=(6,))
+        constrs = [0 <= x, G @ x == h]
+        objective_expression = c @ x
+        prob = cl.Problem(cl.MIN, objective_expression, constrs)
+        Problem.solve(prob, None)
+        # self.assertRaises(RuntimeError, Problem.solve, prob, None)
+
+    def test_parse_integer_constraints(self):
+        x = Variable(shape=(3,), name='my_name_x')
+        y = Variable(shape=(3,), name='my_name_y')
+        z = Variable(shape=(3,), name='my_name_z')
+        invalid_lst = [2, 4, 6]
+        self.assertRaises(ValueError, Problem._parse_integer_constraints, x, invalid_lst)
+        valid_lst = [x, y, z]
+
+        # random problem data
+        G = np.random.randn(3, 6)
+        h = G @ np.random.rand(6)
+        c = np.random.rand(6)
+        # input to coniclift's Problem constructor
+        x = cl.Variable(shape=(6,))
+        constrs = [0 <= x, G @ x == h]
+        objective_expression = c @ x
+        prob = cl.Problem(cl.MIN, objective_expression, constrs)
```

Comment:
It looks like you aren't using the problem data ``G, h, c`` as part of the tests. In this case, a better approach would be to create a trivial problem like
```
prob = cl.Problem(cl.MIN, cl.sum(x), [x==1,  y == 0, z == -1])
```
*without* the re-definition of ``x`` at line 39. If you do this, you might also be able to get away with not manually setting ``prob.variable_map`` from lines 43 -- 46.

Reply:
Alternatively, you could define the problem
```
prob = cl.Problem(cl.MIN, cl.sum(x), [x==1,  y == 0, z[:-1] == -1])
```
which would then result in an error if you tried to set ``z`` to be integer-constrained (because the final component of ``z``, ``z[-1]``, would not appear in the continuous model).

Reply:
Another good test would be to call 
```
z_part = z[:-1]
prob._parse_integer_constraints([x, y, z_part])
```
which should result in a ValueError, because ``z_part`` is not a "proper" Variable object.

Reply:
Will modify the test to define a trivial problem, and include the test you mentioned!